molgenis-data/src/main/java/org/molgenis/data/Fetch.java

```diff
@@ -19,10 +20,22 @@
  */
 public class Fetch implements Iterable<Entry<String, Fetch>> {
   private final Map<String, Fetch> attrFetchMap;
+  private final boolean validated;
 
   /** Creates an empty Fetch. */
   public Fetch() {
+    this(false);
+  }
+
+  /**
+   * Creates an empty valid Fetch. Can be used to improve performance in case of system entity type
+   * fetches defined by the system.
+   *
+   * @param validated whether or not the fetch has been validated
+   */
+  public Fetch(boolean validated) {
```

Comment:
For clarity it's a bit nicer to make this one private and use a factory method instead.